src/System.Private.ServiceModel/tests/Scenarios/Client/ClientBase/ClientBaseTests.cs

```diff
@@ -174,6 +175,163 @@ public static void ClientBaseOfT_Sync_RoundTrip_Call_Using_NetTcpTransport()
+    {
+        string variationDetails = "Client:: BasicHttpBinding/DefaultValues\nServer:: BasicHttpBinding/DefaultValues";
+        string testString = "Hello";
+        StringBuilder errorBuilder = new StringBuilder();
+        bool success = false;
+
+        BasicHttpBinding binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+        binding.TransferMode = TransferMode.StreamedRequest;
+
+        try
+        {
+            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+            IWcfService serviceProxy = factory.CreateChannel();
+
+            Stream stream = StringToStream(testString);
+            success = serviceProxy.UploadStream(stream);
+
+            if (!success)
+            {
+                errorBuilder.AppendLine(String.Format("    Error: expected response from service: '{0}' Actual was: '{1}'", true, success));
+            }
+        }
+        catch (Exception ex)
+        {
+            errorBuilder.AppendLine(String.Format("    Error: Unexpected exception was caught while doing the basic echo test for variation...\n'{0}'\nException: {1}", variationDetails, ex.ToString()));
+        }
+
+        Assert.True(errorBuilder.Length == 0, "Test case FAILED with errors: " + errorBuilder.ToString());
+    }
+
+    [Fact]
+    [OuterLoop]
+    public static void DefaultSettings_Echo_RoundTrips_String_StreamedResponse()
+    {
+        string variationDetails = "Client:: BasicHttpBinding/DefaultValues\nServer:: BasicHttpBinding/DefaultValues";
+        string testString = "Hello";
+        StringBuilder errorBuilder = new StringBuilder();
+        bool success = false;
+
+        BasicHttpBinding binding = new BasicHttpBinding(BasicHttpSecurityMode.None);
+        binding.TransferMode = TransferMode.StreamedResponse;
+
+        try
+        {
+            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(binding, new EndpointAddress(Endpoints.HttpBaseAddress_Basic));
+
+            IWcfService serviceProxy = factory.CreateChannel();
+
+            var returnStream = serviceProxy.GetStream(testString);
+            var result = StreamToString(returnStream);
+            success = string.Equals(result, testString);
+
+            if (!success)
+            {
+                errorBuilder.AppendLine(String.Format("    Error: expected response from service: '{0}' Actual was: '{1}'", true, success));
+            }
+        }
+        catch (Exception ex)
+        {
+            errorBuilder.AppendLine(String.Format("    Error: Unexpected exception was caught while doing the basic echo test for variation...\n'{0}'\nException: {1}", variationDetails, ex.ToString()));
+        }
+
+        Assert.True(errorBuilder.Length == 0, "Test case FAILED with errors: " + errorBuilder.ToString());
+    }
+
+    [Fact]
+    [OuterLoop]
+    public static void DefaultSettings_Echo_RoundTrips_String_Streamed()
+    {
```

Comment:
Thanks for clarifying this. I thought it would be better to make the name indicate the TransferMode being used because we can roundtrip streams even with TransferMode.Buffered.